backend/src/main/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhir.java

```diff
@@ -356,7 +345,9 @@ private String getTestResultSnomed(String input) {
 
   private String getSpecimenTypeSnomed(String input) {
     if (input.matches(ALPHABET_REGEX)) {
-      return specimenTypeToSnomedMap.get(input.toLowerCase());
+      return resultsUploaderDeviceValidationService
+          .getSpecimenTypeNameToSNOMEDMap()
+          .get(input.toLowerCase());
```

Comment:
Is this necessary with `translateSpecimenNameToSNOMED` piece above?  I thought we've already changed the input using the snomed map.  Am I misreading something?

Reply:
I guess that's true! I've been trying to articulate my case and the best I have is: I'm inclined to build FHIR bundles the clean way and keep the impact space of ugly string manipulation & CSV hacking as small as possible.

Like, if we decided we wanted to alter _another_ CSV field in-place, I would think to get rid of `translateSpecimenNameToSNOMED` entirely and build out something more like the FHIR converter for CSV files. This feels like it keeps us on a cleaner two-prong path.

Reply:
That makes sense I think?  

What I had in my head as a future state was a world where we have a single step where we "ingest" the csv (i.e. the api that the user is exposed to) as a first step and when we're done with that we have a proper java object that we use to build the thing (either csv or fhir bundle) we send to report stream.  In my head manipulating fields like this and attaching the processing code would happen as a part of ingestion (as well as validation).  That potential future state is I think why it makes more sense in my head to apply the this change to the file once when it comes in the door, rather than the alternative.

But I'm fine with either approach.